### DISCUSSION TITLE
My Daily Note Toolbar

### DISCUSSION THREAD

**Discussion by chrisgurney** [chrisgurney/obsidian-note-toolbar#17] [Show and tell]
I thought I'd share the current iteration of my Daily Note toolbar, as inspiration for others, and to hopefully to inspire discussion about what you have in yours.

![CleanShot 2024-04-04 at 14 53 44@2x](https://github.com/chrisgurney/obsidian-note-toolbar/assets/17337703/c860e288-5108-4095-a995-6f5358b7eb48)

| Item | What is it? | Why? |
| --- | --- | --- |
| ← | Navigates to my previous daily note via the `Daily notes: Open previous daily note` command. | So I can see where I left off on my work yesterday, or just move through the week's notes. |
| To Do | Inserts today's tasks, from [Things](https://culturedcode.com/things/) using the [Shell Commands](https://github.com/Taitava/obsidian-shellcommands) plugin, running my [things2md](https://github.com/chrisgurney/things2md) Python script. | So I can stay within Obsidian, and work on each task. |
| Focus | Inserts specially-tagged tasks, using the same setup. I use this for daily standup updates. | So I can focus in on what's important today. |
| Events | Inserts headings for today's events, using [icalBuddy](https://hasseg.org/icalBuddy/). | So I have placeholders for meeting notes throughout the day, and I can plan for each. |
| Insert | Pops open a template selector via [Templater](https://github.com/SilentVoid13/Templater)'s `Open Insert Template modal` command. | So I can insert template snippets, such as self-reflection and pre-built Dataview queries. |
| Blocked? | Links to a note I keep updated with strategies to help keep me productive throughout the day. | As mentioned. |
| Done | Inserts today's _completed_ tasks, from [Things](https://culturedcode.com/things/) using the [Shell Commands](https://github.com/Taitava/obsidian-shellcommands) plugin, running my [things2md](https://github.com/chrisgurney/things2md) Python script. | To keep a record of what I worked on. |
| Activity | Inserts today's Git activity (commits, PRs) using the [Shell Commands](https://github.com/Taitava/obsidian-shellcommands) plugin, running my [github2md](https://github.com/chrisgurney/gitactivity2md) Python script. | To keep a record of what I worked on. I also use this to generate the release notes for this plugin. |
| → | Navigates to my _next_ daily note via the `Daily notes: Open next daily note` command. | So I can move through the week's notes. |

On mobile, since I can't run shell commands, I've reduced it primarily to just the navigation options (using the ability to hide items).

Let me know if you have any questions!

**Comment by MrComputron** [comment]
I love this...it's very similar to what I have set up on my daily notes! The insert template modal button is so useful on mobile.

**Comment by heberop** [comment]
Hey  @chrisgurney, could you share your toolbar? I really like it ;) 

Thanks

**Reply by chrisgurney** [reply]
Thanks @heberop! I've added an update [below](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/17#discussioncomment-10927767) in case you haven't seen it.

Let me know if you'd like to use the `things2md` and/or `github2md` scripts I'm using from the Shell Commands plugin, and I can share the command-line arguments for those, as well.

**Reply by heberop** [reply]
Thank you, @chrisgurney. Your work has inspired me to create my own :)
I don't require the scripts at the moment, but I've grasped the concept. I might start on mine soon ;)

**Comment by chrisgurney** [comment]
My Daily Notes toolbar has evolved a little since I posted this, using some of the newer features (separators and menus).

🧩 This toolbar uses the Daily Notes, [Shell Commands](https://obsidian.md/plugins?id=obsidian-shellcommands) and [Status Bar Pomodoro Timer](https://obsidian.md/plugins?id=obsidian-statusbar-pomo) plugins.

![CleanShot 2024-10-13 at 07 30 04@2x](https://github.com/user-attachments/assets/7fa5adc1-50bf-4083-b5dd-c7a767c99855)

How it's changed:

- Grouped note navigation in one section; found it was easier to move between daily notes with the controls being together.
- After _Focus_: Added a shortcut straight to my Things Today list.
- After _Events_: Added a shortcut to open Google Calendar for quick access to my schedule, past today.
- Started using Pomodoro timers. The stopwatch icon starts the [Status Bar Pomodoro Timer](https://obsidian.md/plugins?id=obsidian-statusbar-pomo).
- Grouped common tools that I can use from other toolbars in a _Tools_ menu.

If you're using the v1.14 beta you can import the callout below, but it does use some shell scripts I've written (as noted above), executed by the [Shell Commands](https://github.com/Taitava/obsidian-shellcommands) plugin.

```markdown
> [!note-toolbar|between] Daily Notes
> - [:LiArrowLeft:]()<data data-ntb-command="daily-notes:goto-prev"/> <!-- Previous daily note -->
> - [:LiCalendarCheck:]()<data data-ntb-command="daily-notes"/> <!-- Today's daily note -->
> - [:LiArrowRight:]()<data data-ntb-command="daily-notes:goto-next"/> <!-- Next daily note -->
> - <hr/>
> - [:LiCheckSquare: Focus](<obsidian://shell-commands/?vault%3DDesign%26execute%3Djttqryu9sd>) <!-- Insert tasks to focus on -->
> - [:LiListTodo:](<things:///show?id%3Dtoday>) <!-- Open Things -->
> - [:LiCalendarClock: Events](<obsidian://shell-commands/?vault%3DDesign%26execute%3Dr0qcncyump%26_task_date%3D{{note_title}}>) <!-- Insert events on this day -->
> - [:LiCalendarDays:](<https://calendar.google.com/>) <!-- Open Calendar -->
> - <hr/>
> - [:LiBadgeCheck: Done](<obsidian://shell-commands/?vault%3DDesign%26execute%3Dum5sony1zn%26_task_date%3D{{note_title}}>) <!-- Insert items done on this day -->
> - [:LiGitCommitVertical:](<obsidian://shell-commands/?vault%3DDesign%26execute%3D41451u5h99%26_task_date%3D{{note_title}}>) <!-- Insert Git activity for this day -->
> - [[Atlas/Notes/What to do if I'm Blocked.md|:LiQuestionMarkGlyph:]] <!-- What to do if I'm blocked -->
> - [:LiTimer:]()<data data-ntb-command="obsidian-statusbar-pomo:start-satusbar-pomo"/> <!-- Start Pomodoro timer -->
> - [:LiMenu:]()<data data-ntb-menu="Tools"/> <!-- Tools -->
```
